LDAR_Sim/src/methods/company.py

```diff
@@ -171,58 +173,123 @@ def flag_sites(self):
+                    IFU_sites.append(site)
+                elif len(site['measured_emis_rates']) > self.config['follow_up']['delay']:
+                    target_sites.append(site)
+        else:
+            target_sites = [site for sdx, site in site_wl.items()
+                            if len(site['measured_emis_rates'])
+                            > self.config['follow_up']['delay']]
+
+        # Get portion of sites and threshold filtered sited
+        if self.config['follow_up']['interaction_priority'] == 'proportion':
+            n_sites_to_flag = len(target_sites) * self.config['follow_up']['proportion']
+            n_sites_to_flag = int(math.ceil(n_sites_to_flag))
+            target_sites = [site for site in target_sites[:n_sites_to_flag]
+                            if site['effective_emis_rate']
+                            > self.config['follow_up']['threshold']]
+        else:
+            target_sites = [site for site in target_sites
+                            if site['effective_emis_rate'] > self.config['follow_up']['threshold']]
+            n_sites_to_flag = len(target_sites) * self.config['follow_up']['proportion']
+            n_sites_to_flag = int(math.ceil(n_sites_to_flag))
+            target_sites = target_sites[:n_sites_to_flag]
+
+        # Move IFU Sites to front of queue
+        target_sites = IFU_sites + target_sites
+
+        # Go through all targeted sites and flag them or add them to redundant lists
+        for targ_site in target_sites:
+            site_obj = targ_site['site']
+            site_true_rate = targ_site['effective_emis_rate']
+            venting = targ_site['effective_vent_rate']
+            site_wl.pop(site_obj['facility_ID'])
 
             # If the site is already flagged, your flag is redundant
-            if site['currently_flagged']:
+            if site_obj['currently_flagged']:
                 self.timeseries['{}_flags_redund1'.format(
                     self.name)][self.state['t'].current_timestep] += 1
-
-            elif not site['currently_flagged']:
+            else:
                 # Flag the site for follow up
-                site['currently_flagged'] = True
-                site['date_flagged'] = self.state['t'].current_date
-                site['flagged_by'] = self.config['label']
+                site_obj['currently_flagged'] = True
+                site_obj['date_flagged'] = self.state['t'].current_date
+                site_obj['flagged_by'] = self.config['label']
                 self.timeseries['{}_eff_flags'.format(
                     self.name)][self.state['t'].current_timestep] += 1
 
-                # Does the chosen site already have tagged leaks?
-                redund2 = False
-                for leak in leaks_present:
-                    if leak['date_tagged'] is not None:
-                        redund2 = True
+                # Redand2 is hard to keep track of with multiple sets of serveys
```

Comment:
Probably needs an issue, also should be discussed with the team what these actually mean.

Reply:
Added issue as #114

Reply:
Excellent work, Keegan. If I understand correctly, the purpose of this is to create a watchlist for each company and apply the filter to the watchlist to help only this company's follow-up crew to select sites faster, while better coordinating them so that they always visit the flagged site in order. The logic seems correct to me. My only concern is this additional filter may change the previous follow-up decision-making of LDAR Sim. What will be the difference between this and the previous follow-up on the results?